### ISSUE TITLE
Build Error using common bundle with AMD

### ISSUE THREAD

**Issue by raghu1286** [systemjs/builder#222]
Hi,
I'm using the 0.12.2 release to build a common dependency bundle using "&" then using the generated file to build the main app bundle as shown in the example (https://github.com/systemjs/builder#example---common-bundles), while running the command I'm getting the error 
```
TypeError: Cannot call method 'createElement' of undefined
	Evaluating file:///file_path/build/common.js
	Error loading file:///file_path/build/common.js
```
I'm running the build commands through gulp, the task "common-bundle" runs successfully & the "common.js" file is created, on running the next task "app-bundle" the above error is thrown, below are the task snippets,
```
// Bundle - Common libs
gulp.task('common-bundle', function(cb){
  builder.reset();
  builder.loadConfig('./config.js').then(function() {
      builder.build('app/main & app/modules/account/controller', './build/common.js')
      .catch(error);
  });
});

// Bundle - App
gulp.task('app-bundle', function(cb) {
  builder.reset();
  builder.loadConfig('./config.js').then(function() {
    builder.build('app/main - build/common', './build/app.js')
    .catch(error);
  });
});

```
The jspm config file that I have is,
```
System.config({
  "baseURL": "./",
  "transpiler": "traceur",
  "paths": {
    "*": "*.js",
    "github:*": "jspm_packages/github/*.js",
    "npm:*": "jspm_packages/npm/*.js"
  }
});

System.config({
  "map": {
    "backbone": "npm:backbone@1.2.1",
    "controller": "npm:backbone.controller@0.3.1",
    "jquery": "github:components/jquery@2.1.4",
    "traceur": "github:jmcriffey/bower-traceur@0.0.88",
    "traceur-runtime": "github:jmcriffey/bower-traceur-runtime@0.0.88",
    "underscore": "npm:underscore@1.8.3",
    "process": "npm:process@0.10.1",
    "github:jspm/nodelibs-process@0.1.1": {
      "process": "npm:process@0.10.1"
    },
    "npm:backbone.controller@0.3.1": {
      "backbone": "npm:backbone@1.2.1"
    },
    "npm:backbone@1.2.1": {
      "process": "github:jspm/nodelibs-process@0.1.1",
      "underscore": "npm:underscore@1.8.3"
    }
  }
});
```
Any help is appreciated!

Thanks,
Raghav.

**Comment by guybedford** [comment]
@raghu1286 thanks for reporting.

Try running the common subtraction via `app/main - [build/common]` and let me know if that works out for you?

**Comment by raghu1286** [comment]
Thanks @guybedford  for your quick response.

The suggested subtraction command ```builder.build('app/main - [build/common]', './build/apps.js')``` doesn't throw any errors but it doesn't give the desired output either, the "build/app.js" file has all the dependencies that are present in "build/common.js" as well. The "[ ]" doesn't seem to honor the inputs for exclusion with "-" operator.

Will trying with older release versions make any difference?

Thanks,
Raghav.

**Comment by guybedford** [comment]
@raghu1286 right, yes it should be the first syntax. Can you share the contents of `build/common.js` here for me to take a look?

**Comment by raghu1286** [comment]
The common.js file is available at the url - https://drive.google.com/file/d/0B4Qk6Tfx9_ztWDY0cFFOUGpSMm8/view?usp=sharing

**Comment by guybedford** [comment]
Thanks - it seems this is due to the AMD definition not being able to be evaluated when running in Node. Will put some thought to this use case, will be a bit tricky. A workaround may be to shim jQuery as a global module, which will likely support this use case.

**Comment by raghu1286** [comment]
I've made the common bundling work using the example - https://github.com/systemjs/builder#example---direct-trace-api . But definitely the above approach would be a lot less cleaner & simpler then this.

Thanks for your time & effort for looking into it. 

-Raghav.

**Comment by guybedford** [comment]
I think the thing to do here would be to disable execution of bundles for the build. It is not possible anyway to support bundle addition so I think we just skip this use case.

**Comment by guybedford** [comment]
One solution is simply to have SystemJS detect `"meta"` as a truthy value, and then output `"bundle";` at the top of all bundles. This immediately informs the builder that it doesn't need to execute and can trace the bundle modules, and avoids the issue of an AMD module at the top of the bundle having the bundle not being detected correctly.

**Comment by guybedford** [comment]
I've included a fix for this in https://github.com/systemjs/builder/commit/66554332f0afe6a9d0d485b37cfcfab9be0f548a. If you're able to test it out that would be a great help, but I understand its been a little while!

**Comment by guybedford** [comment]
Released in 0.13.0.
